src/utils.tsx

```diff
@@ -59,25 +73,7 @@ export function highlightSubstring(string, substring) {
   );
 }
 
-export function getClassNames(mixedInput, optionalCondition, output = '') {
-  if (typeof optionalCondition !== 'undefined' && optionalCondition !== true) {
-    return '';
-  }
-
-  if (typeof mixedInput === 'string') {
-    return mixedInput;
-  }
-
-  if (Array.isArray(mixedInput)) {
-    if (typeof mixedInput[1] === 'boolean') {
-      return getClassNames(mixedInput[0], mixedInput[1], output);
-    }
-
-    return mixedInput
-      .map((item) => getClassNames(item, optionalCondition, output))
-      .filter((x) => x) // removes nulls, undefineds, empty strings, etc.
-      .join(' ');
-  }
-
-  return output;
-}
+export const tidyUrlString = (url: string) => {
+  const re = /(^(https?:\/\/)?(www\.)?)|(\/+$)/g;
```

Comment:
2 things:
- Maybe put it outside the function, so that it's compiled once and reused.
- Should URLs like `//:www.ebi.ac.uk/` be cleaned too?

Reply:
- Yep, first point incorporated.
- `//:www.ebi.ac.uk/` are you thinking to handle the `//:` at the start or if we should be leaving `www` there?

Reply:
I'm thinking it would be useful to use https://www.npmjs.com/package/valid-url here to complain in the console so it will be picked up by our error logging service that we have set up

Reply:
I was think `//:www.ebi.ac.uk/` should also become `ebi.ac.uk`, not just for `https//:www.ebi.ac.uk/`

Reply:
I'm not too keen on adding a library for something that will be just used for content that we generate and have control over. Also, I'm more worried about dead links than invalid links in the context of the links we are generating, and this wouldn't catch it

Reply:
* `//:www.ebi.ac.uk/` isn't a valid url (at least it won't open anything in the browser) so the link itself would be broken so not sure if we need to try to handle formatting invalid urls. I am currently handling multiple trailing slashes just because that seems to work in browsers eg `https://wwwdev.ebi.ac.uk////`.
* Yeah deadlinks are a concern but I would also imagine an invalid external link that we provide on the site would want to be detected and fixed if say the url from the API has a typo. As we have so many external links throughout the site my thought is this would be an easy way to detect them as users view them on a page. This can point can be reassessed in the future post alpha.

Reply:
Sorry, I realise I mixed up the URL at the beginning and copy pasted afterwards without realising. I meant `//www.ebi.ac.uk/`

Reply:
This is now handled